test/versioned/undici/requests.test.js

```diff
@@ -426,3 +426,31 @@ test('Undici request tests', async (t) => {
     })
   })
 })
+
+test('Undici error reporting tests', async (t) => {
+  const agent = helper.instrumentMockedAgent({
+    feature_flag: {
+      undici_error_tracking: false
+    }
+  })
+  const undici = require('undici')
+
+  t.after(() => {
+    helper.unloadAgent(agent)
+  })
+
+  await t.test('should not report undici errors', async () => {
+    await helper.runInTransaction(agent, async (tx) => {
+      try {
+        await undici.request('https://invalidurl', {
+          path: '/foo',
+          method: 'GET'
+        })
+      } catch (e) {
+        assert.equal(e.message, 'getaddrinfo ENOTFOUND invalidurl')
```

Comment:
This error message varies in CI. I will add a follow up PR to move this test and document feature flag.

```suggestion
        assert.ok(e)
        assertSegments(tx.trace, tx.trace.root, ['External/invalidurl/foo'], { exact: false })
        assert.equal(tx.exceptions.length, 0)
        tx.end()

```